src/github/services/dead-letter-queue.service.ts

```diff
@@ -0,0 +1,96 @@
+import {Injectable} from '@angular/core';
+import {ConfigService} from '../../app/config.service';
+import {ActionItem, DeadLetterQueue} from '../../domain/action-item';
+import {DEAD_LETTER_QUEUES, QueueConfiguration} from './dead-letter-queues';
+import {BBAuth} from '@blackbaud/auth-client/';
+import {HttpClient} from '@angular/common/http';
+
+export interface DeadLetterQueueReport {
+  service: string;
+  scs: string;
+  zone: string;
+  populated: boolean;
+  url: string;
+  call_failed: boolean;
+}
+
+export interface StatusReport {
+  positive_report: boolean;
+}
+
+@Injectable()
+export class DeadLetterQueueService {
+
+  constructor(private http: HttpClient,
+              private configService: ConfigService) {}
+
+  getActionItems(): Promise<ActionItem[]> {
+    return Promise.all(this.deadLetterQueueReports)
+      .then(reports =>
+        reports.filter(report => report.populated || report.call_failed)
+          .map(populatedReport => new DeadLetterQueue(populatedReport)));
+  }
+
+  private get deadLetterQueueReports(): Promise<DeadLetterQueueReport>[] {
+    return this.queuesToCheck.map(requestData => {
+      return requestData.then(data => {
+        return {
+          restCall: this.getStatusReport(data.url),
+          requestData: data
+        };
+      });
+    }).map(apiRequest => apiRequest.then(request => {
+      return request.restCall.then(statusReport => {
+        return this.convertToReport(request.requestData, statusReport.positive_report);
+      }).catch(() => {
+        return this.convertToErrorReport(request.requestData);
+      });
+    }));
+  }
+
+  private get queuesToCheck(): Promise<DeadLetterQueueReport>[] {
+    return [].concat(...this.queuesConfigurations.map((config) => {
```

Comment:
i think this'll blow up if my team doesn't have any entries in the DLQ file

Reply:
addressed inside `this.queuesConfigurations`